logging/logmanager/plugin_impl_log_manager.go

```diff
@@ -216,3 +229,87 @@ func stringToLogLevel(level string) logging.LogLevel {
 
 	return logging.InfoLevel
 }
+
+// list of supported hook services to send errors to remote syslog server
+const (
+	HookSysLog   = "syslog"
+	HookLogStash = "logstash"
+	HookFluent   = "fluent"
+)
+
+// HookConfig contains configuration of hook services
+type HookConfig struct {
+	Protocol string
+	Address  string
+	Port     int
+	Levels   []string
+}
+
+// commonHook implements that Hook with own level definition
+type commonHook struct {
+	logrus.Hook
+	levels      []logrus.Level
+}
+
+// Levels overrides implementation from embedded interface
+func (cH *commonHook) Levels() []logrus.Level {
+	return cH.levels
+}
+
+// store hook into registy for late use and applies to existing loggers
+func (lm *Plugin) addHook(hookName string, hookConfig HookConfig) error {
+	var lgHook logrus.Hook
+	var err error
+
+	switch hookName {
+	case HookSysLog:
+		var address = hookConfig.Address
+		if hookConfig.Address != "" {
+			address = address + ":" + strconv.Itoa(hookConfig.Port)
+		}
+		lgHook, err = lgSyslog.NewSyslogHook(
+			hookConfig.Protocol,
+			address,
+			syslog.LOG_INFO,
+			"")
+	case HookLogStash:
+		lgHook, err = logrustash.NewHook(
+			hookConfig.Protocol,
+			hookConfig.Address+":"+strconv.Itoa(hookConfig.Port),
+			"vpp-agent")
+	case HookFluent:
+		lgHook, err = logrus_fluent.NewWithConfig(logrus_fluent.Config{
+			Host: hookConfig.Address,
+			Port: hookConfig.Port,
+		})
+	default:
+		return fmt.Errorf("unsupported hook")
+	}
+	if err != nil {
+		lm.Log.Warnf("couldn't create hook for %v : %v", hookName, err.Error())
```

Comment:
This will print two warnings when error occurs. You can rather return custom error: `return fmt.Errorf("creating hook for %v failed: %v", hookName, err)`